base_layer/wallet/src/ffi.rs

```diff
@@ -157,16 +187,27 @@ pub unsafe extern "C" fn add_pending_transaction_outputs(
     pending_tx: *mut PendingTransactionOutputs,
 ) -> bool
 {
+    if wallet.is_null() {
+        return false;
+    }
+    if pending_tx.is_null() {
+        return false;
+    }
+    return true;
     // append this data to the wallet
 }
 
 /// Initialize a Transaction struct to be populated
 #[no_mangle]
 pub unsafe extern "C" fn create_transaction(
-    tx_id: c_ulonglong,    // u64
-    offset: *const c_char, // Byte[32] - PrivateKey
+    tx_id: c_ulonglong,  // u64
+    offset: *mut c_char, // Byte[32] - PrivateKey
 ) -> *mut Transaction
 {
+    let s = CString::from_raw(offset).to_str().unwrap().to_owned();
```

Comment:
As the comment shows this will be a Byte array, not a string, so how will this work in that case?
All PrivateKeys, PublicKeys, Signatures, Proofs, Hashes and Commitments will be a Bytes[32] array with no null terminator.

Reply:
It was just so that it compiles, will be submitting a pr shortly that fixes this and greatly fleshes out the rest.

Hopefully we can finish off the wallet later and then finalize all this.